### PR TITLE
Added GCS access utility module.

### PR DESCRIPTION
**Pull request by dlibenzi**
Dropping this here, until the day it will come to land in pytorch.

### CODE REVIEW

File: torch_xla/utils/gcsfs.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+#
+#  https://cloud.google.com/storage/docs/reference/libraries
+#
+
+import collections
+import io
+import os
+import re
+import tempfile
+from google.cloud import storage as gcs
```
Comment (ailzhang) [inline torch_xla/utils/gcsfs.py]: Maybe we could wrap this in a `try...catch` block and give an error message with `pip install --upgrade google-cloud-storage`?

Comment (dlibenzi) [reply torch_xla/utils/gcsfs.py]: Good idea!

File: torch_xla/utils/gcsfs.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  """Removes a GCS blob.
+
+  Args:
+    path (string): The GCS path of the file. Must be "gs://BUCKET_NAME/PATH"
+      where ``BUCKET_NAME`` is the name of the GCS bucket, and ``PATH`` is a `/`
+      delimited path.
+  """
+  m = re.match(r'gs://([^/]+)/(.+)', path)
+  if not m:
+    raise ValueError('GCS invalid path: {}'.format(path))
```
Comment (zcain117) [inline torch_xla/utils/gcsfs.py]: Mention this exception in the `Raises:` section of the docstring

File: torch_xla/utils/gcsfs.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    path (string): The GCS path of the file. Must be "gs://BUCKET_NAME/PATH"
+      where ``BUCKET_NAME`` is the name of the GCS bucket, and ``PATH`` is a `/`
+      delimited path.
+
+  Returns:
+    A list of ``GcsBlob`` object, having ``path`` and ``size`` fields.
+  """
+  m = re.match(r'gs://([^/]+)(.*)', path)
+  if not m:
+    raise ValueError('GCS invalid path: {}'.format(path))
```
Comment (zcain117) [inline torch_xla/utils/gcsfs.py]: Mention this exception in the Raises: section of the docstring

File: torch_xla/utils/gcsfs.py (hunk 2 above)
Comment (dlibenzi) [reply torch_xla/utils/gcsfs.py]: SG!

**Pull request merged**